samples/cpp/speech_sample/main.cpp

```diff
@@ -205,7 +205,8 @@ int main(int argc, char* argv[]) {
             }
             if (!FLAGS_m.empty()) {
                 for (size_t i = 0; i < outputs.size(); i++) {
-                    model->add_output(outputs[i], ports[i]);
+                    auto newOutput = model->add_output(outputs[i], ports[i]);
+                    newOutput.get_tensor().set_element_type(ov::element::f32);
```

Comment:
it fixes the particular issue because the original network precisions were changed in preprocessing step - [const ov::Layout tensor_layout{"NC"};](https://github.com/mryzhov/openvino/blob/c4ce6c5430de47f71f4e5c03a1be8113e1f2b628/samples/cpp/speech_sample/main.cpp#L88-L96)

So the best fix is to move preprocessing steps right before the model compiling.

@mgrdv @antonrom23  what is the purpose of the preprocessing steps in the speech sample? I think this functionality can be demonstrated in a separate sample

Reply:
I've tried to change the order of preprocessing step and it caused another issue -  convert node (fp16->fp32) created, so the current solution is better.